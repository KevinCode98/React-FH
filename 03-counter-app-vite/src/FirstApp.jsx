import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, numberUser }) => {
    return (
        <>
            <h1 data-testid="test-title"> { title } </h1>
            {/*<code> {JSON.stringify(newMessage)} </code>*/ }
            <p>{ subtitle }</p>
            <p>{ subtitle }</p>
            <p>Your id is: { 100 }</p>
        </>
    );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    // title: "No hay titulo! :c",
    subtitle: "No hay subtitulo! :c",
    name: "Kevin Carrillo"
};