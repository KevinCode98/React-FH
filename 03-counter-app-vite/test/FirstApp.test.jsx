import { FirstApp } from "../src/FirstApp.jsx";
import { render } from "@testing-library/react";

describe("Pruebas en <FirstApp />", () => {
    // test("Debe de hacer match  con el snapshot", () => {
    //     const title = "Mi primera App";
    //     const { container } = render(<FirstApp title={ title }/>);
    //
    //     expect(container).toMatchSnapshot();
    // });

    test("Debe de mostar el titulo en un h1", () => {
        const title = "Mi primera App";
        const { container, getByText, getByTestId } = render(<FirstApp title={ title }/>);

        expect(getByText(title)).toBeTruthy();
        // const h1 = container.querySelector("h1");
        // expect(h1.innerHTML).toContain(title);

        expect(getByTestId("test-title")).toBeTruthy();
        expect(getByTestId("test-title").innerHTML).toContain(title);
    });

    test("Debe de mostrar el subtitulo enviado por props", () => {
        const title = "Mi primera App";
        const subtitle = "Este es el subtitulo";
        const { getAllByText } = render(
            <FirstApp title={ title } subtitle={ subtitle }/>
        );

        expect(getAllByText(subtitle).length).toBe(2);
    });
});