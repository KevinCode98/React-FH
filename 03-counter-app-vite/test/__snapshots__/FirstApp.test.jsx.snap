// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirstApp /> Debe de hacer match  con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Mi primera App
     
  </h1>
  <p>
    No hay subtitulo! :c
  </p>
  <p>
    No hay subtitulo! :c
  </p>
  <p>
    Your id is: 
    100
  </p>
</div>
`;
